gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -638,242 +654,321 @@ _gst_tensor_filter_transform_validate (GstBaseTransform * trans,
-    return GST_FLOW_ERROR;
+  if (!trans_data) {
+    ml_loge ("Failed to allocate memory for internal data of tensor filter transform input data.");
+    return NULL;
   }
 
-  /* 1. Get all input tensors from inbuf. */
-  /* Internal Logic Error or GST Bug (sinkcap changed!) */
-  num_tensors = gst_tensor_buffer_get_count (inbuf);
+  trans_data->num_tensors = gst_tensor_buffer_get_count (buf);
 
-  for (i = 0; i < num_tensors; i++) {
-    in_mem[i] = gst_tensor_buffer_get_nth_memory (inbuf, i);
-    if (!gst_memory_map (in_mem[i], &in_info[i], GST_MAP_READ)) {
+  for (i = 0; i < trans_data->num_tensors; i++) {
+    trans_data->mem[i] = gst_tensor_buffer_get_nth_memory (buf, i);
+    if (!gst_memory_map (trans_data->mem[i], &trans_data->info[i],
+            GST_MAP_READ)) {
       ml_logf_stacktrace
           ("gst_tensor_filter_transform: For the given input buffer, tensor-filter (%s : %s) cannot map input memory from the buffer for reading. The %u-th memory chunk (%u-th tensor) has failed for memory map.\n",
           prop->fwname, TF_MODELNAME (prop), i, i);
-      goto mem_map_error;
+      g_free (trans_data);
+      return NULL;
     }
 
+    trans_data->is_flexible =
+        gst_tensor_pad_caps_is_flexible (GST_BASE_TRANSFORM_SINK_PAD (trans));
     hsize = 0;
-    if (in_flexible) {
-      gst_tensor_meta_info_parse_header (&in_meta[i], in_info[i].data);
-      hsize = gst_tensor_meta_info_get_header_size (&in_meta[i]);
-      gst_tensor_meta_info_convert (&in_meta[i], &prop->input_meta.info[i]);
+    if (trans_data->is_flexible) {
+      gst_tensor_meta_info_parse_header (&trans_data->meta[i],
+          trans_data->info[i].data);
+      hsize = gst_tensor_meta_info_get_header_size (&trans_data->meta[i]);
+      gst_tensor_meta_info_convert (&trans_data->meta[i],
+          &prop->input_meta.info[i]);
     }
 
-    in_tensors[i].data = in_info[i].data + hsize;
-    in_tensors[i].size = in_info[i].size - hsize;
+    trans_data->tensors[i].data = trans_data->info[i].data + hsize;
+    trans_data->tensors[i].size = trans_data->info[i].size - hsize;
   }
 
-  /* 1.1 Prepare tensors to invoke. */
+  return trans_data;
+}
+
+/**
+ * @brief Internal function to get invoke tensors.
+ */
+static GstTensorMemory *
+_gst_tensor_filter_transform_get_invoke_tensors (GstBaseTransform * trans,
+    FilterTransformData * trans_data)
+{
+  GstTensorFilter *self = GST_TENSOR_FILTER_CAST (trans);
+  GstTensorFilterPrivate *priv = &self->priv;
+  GstTensorFilterProperties *prop = &priv->prop;
+  guint i;
+  gsize expected;
+  GstTensorMemory *invoke_tensors = NULL;
+  guint invoke_num_tensors = 0;
+
   if (priv->combi.in_combi_defined) {
+    invoke_num_tensors = g_list_length (priv->combi.in_combi);
```

Comment:
The ```priv->combi.in_combi``` value is set as a property. I'm wondering if a null check is needed.

Reply:
If it's `NULL`, it's returned at L766.